builder.md

```diff
@@ -1,57 +1,180 @@
-# Builder Guidelines
-**Version 0.10.0**
+# Base16 builder specification
+**Version 0.11.0**
 
-A base16 builder is an application that can build syntax highlighting definition files for text editors by using base16 scheme files which contain a collection of colours and base16 template files which contain syntax highlighting rules.
+*The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).*
 
-Builders are designed for theme maintainers' ease of use. Theme maintainers should provide built versions of their theme so the end user doesn't need to be aware of the builder.
+## 1. Introduction
 
-## File Structure
+"Building" refers to replacing placeholders on base16 _[template
+files](./file.md)_ with colors from base16 _[schemes](./styling.md)_.
 
-### Schemes Repository
+A template is a mustache file that acts as a "blueprint": it represents how to
+output the scheme into other file formats, or (more commonly) how to output a
+configuration file to theme a specific software. For example: a [vim
+template](https://github.com/base16-project/base16-vim/blob/main/templates/default.mustache).
 
-The schemes repo should either be stored in a common location (perhaps referred to by environment variable or command line flag) or dynamically embedded in the builder.
+A scheme is a yaml file that represents a palette of 16 colors: to be replaced
+on the placeholders the template files hold. For example: the [solarized
+scheme](https://github.com/base16-project/base16-schemes/blob/main/solarized-dark.yaml)
 
-- `/*.yaml`
+A base16 builder is, essentially, an application that somehow implements the
+_building_ feature specification: filling the placeholders on templates with
+scheme colors.
 
-### Template Repository
+Builders are categorized into:
```

Comment:
I'd like to specify somewhere that there are 2 levels of implementing the spec:

1. the templates and variables spec
2. the cli

Both should be considered "spec compliant" builders, but the CLI is more specific because they need to implement both specs to be fully compliant.

Reply:
Agree, that seems better